First/Entities/FighterRecord.cs

```diff
@@ -8,8 +8,21 @@ public class FighterRecord
         public int KOs { get; set; }
         public int Draws { get; set; }
 
-        public double Rank { get; set; }
-        public double PreviousRank { get; set; }
+        private double _rank;
+
+        public double Rank {
+            get { return _rank; }
+            set { PreviousRank = _rank; _rank = value; }
+            }
+
+        public double PreviousRank { get; private set; } // automatically preserved
```

Comment:
You are setting previousRank in Rank but also have a setter in PreviousRank (line 18)? Why do you need both?
I am not sure how it works 100%, just asking maybe you know this better

Reply:
I have to use a private setter for Previous Rank to use it in the Rank setter. Alternatively, I could have a separate _previousRank private variable.

Reply:
set { **PreviousRank = _rank**; _rank = value; }
            }

        public double PreviousRank { get; private set; }